docs/madr/decisions/055-meshservice-subset.md

```diff
@@ -0,0 +1,175 @@
+# Subsets of Dataplanes per MeshService
+
+* Status: accepted
+
+## Context and Problem Statement
+
+Referring to a subset of endpoints for a given Service is important for service
+mesh. It's most commonly use to implement things like blue/green deployments or canary
+deployments.
+
+For example if we have a `backend` service that we want to upgrade from `v1` to
+`v2`. With `kuma.io/service`, this would mean having `Dataplanes` with
+`kuma.io/service: backend` and an additional tag either `version: v1` or
+`version: v2`. In order to do weighted routing to these different versions, a
+user would create a `MeshHTTPRoute`:
+
+```yaml
+        backendRefs:
+        - kind: MeshServiceSubset
+          tags:
+            version: v2
+          name: backend
+          weight: 10
+        - kind: MeshServiceSubset
+          tags:
+            version: v1
+          name: backend
+          weight: 90
+```
+
+Now that we have `MeshService` objects, the user has something like the following:
+
+```yaml
+kind: MeshService
+spec:
+  selector:
+    dataplaneTags:
+      app: backend
+```
+
+and matched `Dataplanes` have `version` tags set.
+
+The question this MADR answers is how do users handle the same concept with
+`MeshService`?
+
+## Considered Options
+
+* Support `MeshServiceSubset` with `MeshService` objects
+* Users create additional `MeshServices`
+
+## Decision Outcome
+
+Chosen option: additional `MeshServices`, no more `MeshServiceSubset`
+
+Concretely:
```

Comment:
shouldn't all of this up to `### `MeshServiceSubset` support` be in the `### Additional `MeshService` support` section?

Reply:
I don't know. I mean it's the section that defines which is the decided outcome. This way the user can see the way it will work. And if they want to look at other alternative they can go to that scetion. If you really want me to move it I can, I find it easier to read this way

Reply:
It's easier for me when you lay out options and you say which one you pick. But I don't mind.